packages/hydrooj/src/model/solution.ts

```diff
@@ -72,6 +73,8 @@ class SolutionModel {
     static async vote(domainId: string, psid: ObjectId, uid: number, value: number) {
         let pssdoc = await document.getStatus(domainId, document.TYPE_PROBLEM_SOLUTION, psid, uid);
         await document.setStatus(domainId, document.TYPE_PROBLEM_SOLUTION, psid, uid, { vote: value });
+        const { owner } = await SolutionModel.get(domainId, psid);
+        await domain.incUserInDomain(domainId, owner, 'nLike', (value === 1 ? 1 : 0) - (pssdoc && pssdoc.vote === 1 ? 1 : 0));
```

Comment:
If you want to add it, please clean it when delete a solution.

Reply:
Should I calculate it together with upvote to improve efficiency?